LeanCamCombi/PlainCombi/OrderShatter.lean

```diff
@@ -44,9 +44,13 @@ def OrderShatters : Finset (Finset α) → List α → Prop
       ∧ ∀ ⦃s : Finset α⦄, s ∈ 𝒜.nonMemberSubfamily a → ∀ ⦃t⦄, t ∈ 𝒜.memberSubfamily a →
         {x ∈ s | a < x} = {x ∈ t | a < x}
 
-instance : DecidablePred 𝒜.OrderShatters
-  | [] => decidableNonempty
-  | a :: l => by unfold OrderShatters; sorry
+instance : DecidablePred 𝒜.OrderShatters := by
+  intro l
+  revert 𝒜
+  induction l
+  · exact decidableNonempty
+  · intro
+    apply instDecidableAnd
```

Comment:
I was surprised that your proof worked when my stub didn't. Turns out that the instance you're defining by pattern-matching is unknown to typeclass search until the end of the definition.

Here is a slightly more idiomatic (and shorter) way of writing your proof. Note the `generalizing 𝒜` instead of the `revert 𝒜`, `intro` pattern.
```suggestion
instance instDecidableRel : DecidableRel (OrderShatters (α := α)) := fun 𝒜 l ↦ by
  induction l generalizing 𝒜
  · exact decidableNonempty
  · exact instDecidableAnd
```